### PR TITLE
EON version binding

### PR DESCRIPTION
**Pull request by ddrvar**
[Jira ticket](https://horizenlabs.atlassian.net/browse/SDK-1207?atlOrigin=eyJpIjoiYWExMDcyNjY4ZGI0NDNlODg1ZWY4ZDFkOGY1ZGI5NzQiLCJwIjoiaiJ9)

EON version is taken from the class package implementation version which is specified in pom.xml file. When running EON from IDE, it returns `null` because the implementation version is populated when JAR is built. It follows the logic of sdk version taken from [here](https://github.com/HorizenOfficial/Sidechains-SDK/blob/7729cae9fd10670e57f02decc418b552906eab0c/sdk/src/main/scala/io/horizen/account/api/rpc/service/RpcUtils.scala#LL16C1-L17C1).

### CODE REVIEW

File: node/src/main/java/io/horizen/eon/EonAppModule.java [hunk 1]
```diff
@@ -96,5 +96,16 @@ public void configureApp() {
         bind(Integer.class)
                 .annotatedWith(Names.named("ConsensusSecondsInSlot"))
                 .toInstance(CONSENSUS_SLOT_TIME);
+
+        bind(String.class)
+                .annotatedWith(Names.named("EonVersion"))
```
Comment (MarcoOl94) [inline node/src/main/java/io/horizen/eon/EonAppModule.java]: Eon -> App

File: node/src/main/java/io/horizen/eon/EonAppModule.java [hunk 2]
```diff
@@ -96,5 +96,16 @@ public void configureApp() {
...
+                .toInstance(getEONVersion());
+    }
+
+    public String getEONVersion() {
+        String defaultVersion = "dev";
```
Comment (MarcoOl94) [inline node/src/main/java/io/horizen/eon/EonAppModule.java:115]: To me it should be better to enforce having a declared version. I would like to always read the version from the pom, but please sync up with @paolocappelletti if this is an acceptable behavior.

**Review by MarcoOl94 — CHANGES_REQUESTED** [review]

File: node/src/main/java/io/horizen/eon/EonAppModule.java (hunk 1 above)
Comment (ddrvar) [reply node/src/main/java/io/horizen/eon/EonAppModule.java]: Done

File: node/src/main/java/io/horizen/eon/EonAppModule.java (hunk 2 above)
Comment (ddrvar) [reply node/src/main/java/io/horizen/eon/EonAppModule.java:115]: Retrieves the EON version by dynamically accessing the implementation version of the package at runtime.
The implementation version is taken from the JAR file's manifest file **Implementation-Version** attribute. That attribute
is populated when JAR is built with the value of the version tag under the project key from the pom file. 
When running the application in a development environment (e.g., directly from the source code or an IDE), the implementation version is not accessible and will return the default "dev" value.

**Review by MarcoOl94 — APPROVED** [review]

**Review by paolocappelletti — APPROVED** [review]

**Pull request merged**